src/httpScript.js

```diff
@@ -238,10 +238,10 @@ function httpRequestDefault(options) {
   return options
 }
 
-export default function httpScript(script, httpRequest = httpRequestDefault) {
+export default function httpScript(script, httpRequest = httpRequestDefault, initialState) {
   const parsedList = parseHttpScript(script)
 
-  let state = {}
+  let state = {initialState}
```

Comment:
I think it could be better to declare state as a parameter and set the default to an empty object